hrm-service/src/contact/contact.ts

```diff
@@ -79,11 +112,31 @@ export const createContact = async (
       newContact.queueName || (<any>(rawJson ?? {})).queueName,
     createdBy,
   };
-  return create(
+
+  if (completeNewContact.taskId) {
+    const existingContact: Contact = await getByTaskId(accountSid, completeNewContact);
+    if (existingContact) {
+      // A contact with the same task ID already exists, return it (idempotence)
+      return existingContact;
+    }
+  }
+
+  const created = await create(
     accountSid,
     completeNewContact,
     (newContact.csamReports ?? []).map(csr => csr.id),
   );
+
+  // TODO: this being not in the same trx, might fail, the contact be created, but the request fail with 5xx. Idempotence prevents inconsistencies but do we want this behavior?
+  if (isChatChannel(created.channel)) {
```

Comment:
I remember this being discussed when we were talking about the design, and maybe I heard what I wanted to but I thought we didn't wanted to lose the contact information in case of a job-creation-failure? I will rollback this changes tho.

Reply:
Whilst there are sometimes good reasons to break it, a good rule of thumb is that en endpoint should either do everything it is supposed to and return a success or do nothing and return an error. Endpoints that can possibly leave their work in a half-completed state just add more complexity to a system, more possible scenarios to account for.

In this case, whilst technically possible, we will almost never see a scenario where the first query gets to run, but the DB is down by the time the second query gets to run microseconds later. This failure scenario is far more likely to be due to broken application or database code, which we want to fail noisily and fix (hoefully in staging ;-) ), rather than leaving half complete contacts around.

Reply:
We don't want to lose contacts in the case of the job itself failing, but I think job creation is better being transactional since it's just another DB operation